File: 08青蛙跳台阶/Solution.java

public class Solution {
    public int JumpFloor(int n) {
        if(n <= 1)
    return n;
    // 先创建一个数组来保存历史数据
    int[] dp = new int[n+1];
    // 给出初始值
    dp[0] = 0;
    dp[1] = 1;
    dp[2]=2;
    // 通过关系式来计算出 dp[n]
    for(int i = 3; i <= n; i++){
        dp[i] = dp[i-1] + dp[i-2];
    }
    // 把最终结果返回
    return dp[n];
    }
}
// 注意初始值和斐波那契数列的不同

File: 08青蛙跳台阶/Solution.cpp
/* 一只青蛙一次可以跳上1级台阶，也可以跳上2级。
求该青蛙跳上一个n级的台阶总共有多少种跳法
（先后次序不同算不同的结果）。 */
class Solution {
public:
    int jumpFloor(int n) {
        if(n <= 2)
    return n;
    
   int  p1=1;
   int p2=2;
   int p;

    for(int i = 3; i <= n; i++){
       p=p1+p2;
       p1=p2;
       p2=p;
    }
    return p;
    }
};
// 可以用动态规划解决 